src/pybind/mgr/dashboard/services/ganesha.py

```diff
@@ -60,42 +69,54 @@ def _get_clusters_locations(cls):
                 else:
                     pool, namespace = pool_nm.split('/', 1)
 
+            if cluster in orch_result:
+                # cephadm might have set some cluster settings, ask the user to remove it.
+                raise NFSException(
+                    'A conflicted NFS-Ganesha cluster name `{0}` is detected. There exists an '
+                    'NFS-Ganesha cluster called `{0}` that is deployed by the Orchestrator. '
+                    'Please remove or rename the cluster from the GANESHA_RADOS_POOL_NAMESPACE '
+                    'setting.'.format(cluster))
```

Comment:
```suggestion
            if cluster in orch_result:
                # cephadm might have set some cluster settings, ask the user to remove it.
                logger.warning(
                    'A conflicted NFS-Ganesha cluster name `{0}` is detected. There exists an '
                    'NFS-Ganesha cluster called `{0}` that is deployed by the Orchestrator. '
                    'Please remove or rename the cluster from the GANESHA_RADOS_POOL_NAMESPACE '
                    'setting.'.format(cluster))
                    del orch_result[cluster]
```

raising feels like would be painful in an upgrade case because it would force the user to migrate their export config into another RADOS pool and manipulate the dashboard config settings?

would it be better to log a warning and default back to the legacy behavior in this case?

Reply:
> -  if the cluster is only partially upgraded, the user would no longer be able to manage any exports
>  - the user would need to migrate all their export configs into another RADOS pool and then remove the dashboard config setting on a per cluster basis ?

If we allow using legacy way to manage a cluster under migration, the changes made during migration only persist in the old pool/namespace (assume new pool/namespace is created for migration). If the migration uses the same pool/namespace, we don't want users to change the objects either (because we are migrating them).

Removing the old setting in the Dashboard after the migration is sufficient, using a different pool/namespace is not mandatory.